resources/assets/js/label-trees/components/labelTrees.vue

```diff
@@ -20,126 +17,151 @@
                 :flat="true"
                 :showFavouriteShortcuts="true"
                 :collapsible="collapsible"
                 @select="handleSelect"
                 @deselect="handleDeselect"
                 @remove-favourite="handleRemoveFavourite"
-                ></label-tree>
+            ></label-tree>
             <label-tree
-                v-for="tree in trees"
+                v-for="(tree, index) in sortedTrees"
                 :key="tree.id"
+                :treeIndex="index"
+                :maxTreeIndex="sortedTrees.length - 1"
                 :name="tree.versionedName"
                 :labels="tree.labels"
                 :multiselect="multiselect"
                 :allow-select-siblings="allowSelectSiblings"
                 :allow-select-children="allowSelectChildren"
                 :show-favourites="showFavourites"
+                :show-custom-order="showCustomOrder"
                 :collapsible="collapsible"
                 @select="handleSelect"
                 @deselect="handleDeselect"
                 @add-favourite="handleAddFavourite"
                 @remove-favourite="handleRemoveFavourite"
-                ></label-tree>
+                @move-label-trees="moveLabelTrees"
+            ></label-tree>
         </div>
     </div>
 </template>
 
 <script>
-import Keyboard from '@/core/keyboard.js';
-import LabelTree from './labelTree.vue';
-import mitt from 'mitt';
-import Typeahead from './labelTypeahead.vue';
-import {MAX_FAVOURITES} from '../constants.js';
+import Keyboard from "@/core/keyboard.js";
+import LabelTree from "./labelTree.vue";
+import mitt from "mitt";
+import Typeahead from "./labelTypeahead.vue";
+import { MAX_FAVOURITES } from "../constants.js";
 
 /**
  * A component that displays a list of label trees.
  *
  * @type {Object}
  */
 export default {
-    emits: [
-        'add-favourite',
-        'clear',
-        'deselect',
-        'remove-favourite',
-        'select',
-    ],
+    emits: ["add-favourite", "clear", "deselect", "remove-favourite", "select"],
     components: {
         typeahead: Typeahead,
-        labelTree: LabelTree,
+        labelTree: LabelTree
     },
     data() {
+        let volumeProjectIds = biigle.$require("volumes.projectIds");
+        let annotationProjectIds = biigle.$require("annotations.projectIds");
+        let largoProjectIds = biigle.$require("largo.projectIds");
+        let largoProject = [biigle.$require("largo.projectId")];
+        let videoProjectIds = biigle.$require("videos.projectIds");
+
+        let projectIds = [volumeProjectIds, annotationProjectIds, largoProjectIds, largoProject, videoProjectIds]
+            .flat()
+
+        let customOrderStorageKeys = [];
+        projectIds.forEach(
+            function (el) {
+                if (Number.isInteger(el)) {
+                    return customOrderStorageKeys.push(`biigle.label-trees.${el}.custom-order`)
+                }
+            }
+        );
```

Comment:
You should add these as a new prop of the component instead of hard-coding the variable names.